### ISSUE TITLE
GlobalCheckpointSyncIT#testBackgroundGlobalCheckpointSync apparently stuck during recovery

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#57660]
**Build scan**:

https://gradle-enterprise.elastic.co/s/nunjhzthd7sam

**Repro line**:

```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.seqno.GlobalCheckpointSyncIT.testBackgroundGlobalCheckpointSync" \
  -Dtests.seed=697F95BFC085C2D7 \
  -Dtests.security.manager=true \
  -Dtests.locale=zh-CN \
  -Dtests.timezone=America/Kentucky/Louisville \
  -Druntime.java=11
```

**Reproduces locally?**:

Not after 50 repeats

**Applicable branches**:

`master`

**Failure history**:

3 failures on `master` since 2020-05-27, the next-most-recent failure of this test was 2019-08-05 and looks unrelated

https://build-stats.elastic.co/goto/0aeca547109980a862e1b4e939832e3c

**Failure excerpt**:

```
org.elasticsearch.index.seqno.GlobalCheckpointSyncIT > testBackgroundGlobalCheckpointSync FAILED
    java.lang.AssertionError: timed out waiting for green state
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:892)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:845)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:834)
        at org.elasticsearch.index.seqno.GlobalCheckpointSyncIT.runGlobalCheckpointSyncTest(GlobalCheckpointSyncIT.java:210)
        at org.elasticsearch.index.seqno.GlobalCheckpointSyncIT.testBackgroundGlobalCheckpointSync(GlobalCheckpointSyncIT.java:99)

    java.lang.AssertionError: All incoming requests on node [node_s2] should have finished. Expected 0 but got 229; pending tasks [[{
      "node" : "node_s2",
      "id" : 225,
      "type" : "transport",
      "action" : "internal:index/shard/recovery/translog_ops",
      "description" : "",
      "start_time" : "2020-06-04T09:06:06.676Z",
      "start_time_in_millis" : 1591261566676,
      "running_time" : "1.6m",
      "running_time_in_nanos" : 98505040596,
      "cancellable" : false,
      "headers" : { }
    }]]
```

Possibly this duplicates #57416.

**Comment by tbrooks8** [comment]
This does not appear to be the same as #57416 to me. #57416 is specific to the disruption infrastructure in that test.

**Comment by tbrooks8** [comment]
@dnhatn recently merged a PR related to this test suite (#57567). Maybe he can clarify if this might be related?

**Comment by tbrooks8** [comment]
I think this is fixed by #57859.

**Comment by DaveCTurner** [comment]
Thanks for investigating @tbrooks8
